components/mintKamon/GenerateImageForm.tsx

```diff
@@ -1,13 +1,215 @@
+        title: 'All fields are required',
+        description: toastDescription,
+        status: 'error'
+      })
+      return
+    }
+    setIsLoading(true)
+    // this is debug purpose
+    // setTokenImageURI(
+    //   'https://pitpa.mypinata.cloud/ipfs/QmNndSLYnChANwhPBnj5AwB5M7XEWz5LwDbQsrXNLu89Gb'
+    // )
+    const blobToBase64 = (blob: any): Promise<string> => {
+      return new Promise<string>((resolve, _) => {
+        const reader = new FileReader()
+        reader.onloadend = () => {
+          if (typeof reader.result == 'string') {
+            resolve(reader.result)
+          } else {
+            resolve('')
+          }
+        }
+        reader.readAsDataURL(blob)
+      })
+    }
+
+    const payLoad = {
+      name,
+      address: data?.address,
+      roles: ['Community member'],
+      points: 100,
+      date: ~~(new Date().getTime() / 1000),
+      profilePicture: await blobToBase64(imageFileObject)
+    }
+
+    let ipfsApiEndpointRequest
+    try{
+      ipfsApiEndpointRequest = await fetch(ipfsApiEndpoint, {
+        body: JSON.stringify(payLoad),
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        method: 'POST'
+      })
+      .then((response) => {
+        // 413 crashes before we can check ipfsApiEndpointRequest.status,
+        // so it's dealt with here
+        if (response.status == 413) throw new Error(response.status.toString())
+        return response
+      })
+    } catch (error) {
+      processHttpError(413)
+      setIsLoading(false)
+      return
+    }
+    processHttpError(ipfsApiEndpointRequest.status)
+    const ipfsApiEndpointResponse = await ipfsApiEndpointRequest.json()
+    const tokenURI = await ipfsApiEndpointResponse.tokenUri
+
+    const pinataTokenUriRequest = await fetch(tokenURI)
+    processHttpError(pinataTokenUriRequest.status)
+    const responseJson = await pinataTokenUriRequest.json()
+    const tokenImageURI = await responseJson.image
+    if (tokenImageURI) {
+      setTokenImageURI(tokenImageURI)
+      return
+    }
+    setIsLoading(false)
+  }
+
+  const processHttpError = (status: number) => {
+    const httpErrorMsd: number = ~~(status/100)
+    const httpErrorMessages = {
+      error4: 'There was a problem with the request',
+      error400: 'There probably was an error with the data you submitted.',
+      error403: 'The server did not allow the request.',
+      error404: 'The server did not know what to do with the request.',
+      error413: 'The request payload was too large. Please choose an image below 30 megabytes.',
+      error5: 'The server could not fulfill the request. Please try again later.',
+      error500: 'The server crashed and could not fulfill the request. Please try again later.',
+    }
+    if (httpErrorMsd > 3) {
+      const propA: string = 'error' + status
+      const propB: string = 'error' + httpErrorMsd
+      let toastDescription: string
+      if (httpErrorMessages.hasOwnProperty(propA)) {
```

Comment:
I implemented 3 and forced most of the errors to see that it works, but `yarn run build` fails and I don't really understand the message. Can anyone help with this?

```
Type error: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ error4: string; error400: string; error403: string; error404: string; error413: string; error5: string; error500: string; }'.
  No index signature with a parameter of type 'string' was found on type '{ error4: string; error400: string; error403: string; error404: string; error413: string; error5: string; error500: string; }'.
```

Reply:
fixed here

https://github.com/henkaku-center/membership-nft-interface/pull/40/files#diff-b5d6854ad4950112e6160319f654951ab1e76bc19dab636b6f6e6a8eed39f0b3R127